### ISSUE TITLE
wsl2: Failed to install 'ms-vscode-remote.remote-wsl'

### ISSUE THREAD

**Issue by imoldfella** [microsoft/vscode-remote-release#3010]
Hoping you can give me some things to try, I'm probably doing something wrong? This worked for a couple weeks, then just stopped connecting after recent windows update. This is a custom kernel, but I haven't had problems with it on anything but vs code. When I use the debug address from bash/wsl I get 6849775440496504e0677ffcf2a46d47ad9130f5. Strangely from windows I can connect to that debug address as localhost, but not as 127.0.0.1. Both localhost and 127.0.0.1 work from ubuntu. Thanks, visual studio code is great.

[log.txt](https://github.com/microsoft/vscode-remote-release/files/4637883/log.txt)

**Comment by Shahaed** [comment]
I am also getting this on WSL2

**Comment by tautomer** [comment]
For me the problem was that the folder `%userprofile%\.vscode\extensions\ms-vscode-remote.remote-wsl-0.44.2` was locked by`wslhost`, so VS Code won't be able to delete the folder and reinstall remote-wsl.

Qutting VS code, killing wsl host and deleting the folder allowed me to reinstall remote-wsl.

**Comment by WSLUser** [comment]
That didn't work for me. I also was using Clear Linux on WSL2 if that makes a difference.

**Comment by Msadr471** [comment]
I am also getting this on WSL2 

what should I do?
![image](https://user-images.githubusercontent.com/49529241/86699240-3f819000-c025-11ea-974d-6790f0831b6b.png)

**Comment by PavelSosin-320** [comment]
Aware of incompatible changes in WSL 2 release since 20151! I Only Distros installed after upgrade started for me smoothly. All others required some manipulations like re-install. It works! It only starts differently! WSL does some things once, during 1st distro start. There is no automatic dependents upgrade.

**Comment by Msadr471** [comment]
> Aware of incompatible changes in WSL 2 release since 20151! I Only Distros installed after upgrade started for me smoothly. All others required some manipulations like re-install. It works! It only starts differently! WSL does some things once, during 1st distro start. There is no automatic dependents upgrade.

thank you. Yes, I was able to find the problem and fix it.

**Comment by haslinghuis** [comment]
Cannot install any extensions due to XHR failure? (At least I could install the WSL Remote Extension manually)

**Comment by haslinghuis** [comment]
Two rules in Defender Firewall were blocking access. Guess a telemetry blocking script I forget I had installed is now begging for attention 😦 
```
telemetry_cs1.wpc.v0cdn.net
telemetry_az361816.vo.msecnd.net
```
Note: It is not about being tracked. I use my smartphone as router with unlimited data plan but still like to minimize traffic because a 5GB day limit with free SMS upgrades which are annoying, so I like to keep my daily traffic below the 5GB limit.

**Comment by aeschli** [comment]
Looking at the logs in the original description, WSL2 is detected but in the end we try to connect it as WSL1.

There has been fixes for that in 0.44.4.

@imoldfella Do you still see the issue?
